portfolio/src/main/webapp/script.js

```diff
@@ -67,5 +70,50 @@ function createListElement(name, text) {
  */
 function deleteCommentData() {
   const request = new Request('/delete-data', {method: POST});
-  fetch(request).then(result => getCommentData());
+  fetch(request).then(response => getCommentData());
+}
+
+/**
+ * Fetch user login status from LoginServlet.
+ */
+function getLoginStatus() {
+  fetch('/login').then(response => response.json()).then((userStatus) => {
+    if (userStatus.isLoggedIn) {
+      document.getElementById('comments-form-div').style.display = 'show';
+      document.getElementById('delete-button-div').style.display = 'show';
+      document.getElementById('handle-login').innerHTML =
+          userStatus.loginMessage;
+    } else {
+      document.getElementById('comments-form-div').style.display = 'none';
+      document.getElementById('delete-button-div').style.display = 'none';
+      const messageElement = document.createElement(COMMENT_CHILD_TAG);
+      messageElement.innerText = userStatus.loginMessage;
+      document.getElementById('handle-login').innerHTML =
+          messageElement.innerText;
+    }
+  });
+}
+
+/**
+ * Fetch URL of user-uploaded image from Blobstore servlet.
+ */
+function fetchBlobstoreUrl() {
+  fetch('/blobstore-upload-url')
+      .then((response) => {
+        console.log(response);
+        return response.text();
+      })
+      .then((imageUploadUrl) => {
+        const messageForm = document.getElementById('comments-form');
+        messageForm.action = imageUploadUrl;
+        messageForm.classList.remove('hidden');
+      });
+}
+
+/**
+ * Calls all functions necessary to initialize page.
+ */
+function initializePage() {
+  getLoginStatus();
+  fetchBlobstoreUrl();
```

Comment:
you're sending two requests and both of these need to be returned before anyone is able to submit the comments-form. will comments-form display before the blobstoreUrl returns?

i'm wondering if it makes more sense to have a single endpoint that provides all the data to display the from (login status and form target).

Reply:
I see- I'll combine the functions into one